django/miracle/core/models.py

```diff
@@ -567,6 +560,14 @@ def __unicode__(self):
         return u'{} (internal: {})'.format(self.full_name, self.name)
 
 
+class DatasetFile(models.Model):
+
+    ignored = models.BooleanField(default=False)
+    datatable = models.ForeignKey(DataTable, null=True, related_name='files')
+    project = models.ForeignKey(Project, related_name='datatablefiles')
```

Comment:
Is there a need for another link to the Project when you can retrieve it via datatable.dataset.project? It is faster but introduces possibility of data degeneracy (i.e., what do you do if project != datatable.dataset.project)